libtransmission/port-forwarding-upnp.cc

```diff
@@ -261,8 +261,13 @@ tr_port_forwarding_state tr_upnpPulse(
 
         FreeUPNPUrls(&handle->urls);
         auto lanaddr = std::array<char, TR_ADDRSTRLEN>{};
-        if (UPNP_GetValidIGD(devlist, &handle->urls, &handle->data, std::data(lanaddr), std::size(lanaddr) - 1) ==
-            UPNP_IGD_VALID_CONNECTED)
+        if (
+#if (MINIUPNPC_API_VERSION >= 18)
+            UPNP_GetValidIGD(devlist, &handle->urls, &handle->data, std::data(lanaddr), std::size(lanaddr) - 1, NULL, 0)
```

Comment:
Maybe consider introducing a small wrapper, i.e.
```cpp
int tr_upnp_get_valid_igd(UPNPDev* devlist, tr_upnp& handle, std::array<char, TR_ADDRSTRLEN>& lanaddr)
{
#if MINIUPNPC_API_VERSION >= 18
    return UPNP_GetValidIGD(devlist, &handle.urls, &handle.data, std::data(lanaddr), std::size(lanaddr) - 1, nullptr, 0);
#else
    return UPNP_GetValidIGD(devlist, &handle.urls, &handle.data, std::data(lanaddr), std::size(lanaddr) - 1);
#endif
}

...

        if (tr_upnp_get_valid_igd(devlist, *handle, lanaddr) == UPNP_IGD_VALID_CONNECTED)
        {
            ...
        }
```

If not, at least
```suggestion
            UPNP_GetValidIGD(devlist, &handle->urls, &handle->data, std::data(lanaddr), std::size(lanaddr) - 1, nullptr, 0)
```